### PR TITLE
fix(nef): NEF lockless review follow-ups

### PR DESCRIPTION
**Pull request by ysndr**
Follow-ups to the merged NEF lockless-resolution stack (#4271, #4291,
#4375, #4401, #4409, #4450, #4456), addressing the code-actionable review
comments from @dcarley. One commit per comment.

## Changes

- **fix(flox-build.mk): quote the --out target expansion** — quote `$@` (#4450)
- **fix(flox-build.mk): quote the catalog-lockfile prerequisite expansion** — quote `$<` (#4450)
- **docs(publish): correct the locked_inputs comment** — always sent, empty
  when none; older CLIs coalesced server-side since floxhub#1791 (#4456)
- **refactor(build): rename catalogLockfile → direct_catalog_inputs** — the
  field is a map of inputs, not a file; internal rename incl. the makefile's
  emitted JSON key (#4456)
- **fix(nef-lock-catalog): default stability to BaseCatalogInfo::DEFAULT_STABILITY** (#4409)
- **fix(nef-lock-catalog): parse stability at the CLI boundary** — fail fast on
  invalid `--stability`; removes the now-dead `InvalidStability` variant (#4401)
- **test(nef-lock-catalog): align success lookup fixture with catalog-server** —
  slash canonical keys, unprefixed matched values, always-present `inputs` (#4401)
- **test(nef-lock-catalog): use wire-form refs in the partial lookup fixture** —
  unprefixed refs; `render.rs` restores the `catalogs.` root (#4401)

## Deferred to Linear

Non-code review follow-ups were filed rather than addressed here:
ECO-133 (scanner parse/IO errors), ECO-134 (scanner edge cases), ECO-135
(wide `catalog.*` warning), ECO-136/137 (NEF test coverage) + ECO-141
(interim fixture scaffolding), ECO-138 (stability semantics + makefile
default), ECO-139 (error-messaging UX), ECO-140 (superseded-locks docs).
Token propagation for the lock libexec is tracked in CLI-112.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by billlevine — APPROVED** [review]
Nice!

File: cli/nef-lock-catalog/src/bin/lock.rs [hunk 1]
```diff
@@ -51,9 +53,11 @@ struct Cli {
     #[arg(long)]
     out: Option<PathBuf>,
 
-    /// Catalog stability channel.
-    #[arg(long, default_value = "stable")]
-    stability: String,
+    /// Catalog stability channel. Parsed at the CLI boundary so an invalid
+    /// value fails fast; the server remains the authority on which stabilities
+    /// exist.
```
Comment (dcarley) [inline cli/nef-lock-catalog/src/bin/lock.rs]: This changed the user facing `--help` output and they probably don't need to know that
```suggestion
    /// Catalog stability channel.
```

File: cli/nef-lock-catalog/test_data/build_inputs_lookup/success.json
Comment (dcarley) [inline cli/nef-lock-catalog/test_data/build_inputs_lookup/success.json:1]: (non-blocking) Realised the JSON fixtures in https://github.com/flox/flox/blob/main/cli/nef-lock-catalog/src/lock/transform.rs also use the wrong format keys names but 🤷🏻

**Review by dcarley — APPROVED** [review]

File: cli/nef-lock-catalog/test_data/build_inputs_lookup/success.json
Comment (ysndr) [reply cli/nef-lock-catalog/test_data/build_inputs_lookup/success.json:1]: fwiw, i dont even know why they need to be keyed in the first place, rather than being a set. cc @billlevine

Comment (billlevine) [reply cli/nef-lock-catalog/test_data/build_inputs_lookup/success.json:1]: They are needed on the lock side to reference DAG edges via the "inputs" field and are used as references in the "matched" field.  Granted they are not used at all in the publish side, and that could be a set, b/c there is no matched, nor are there inputs or DAG edges b/c we only want the direct dependencies.

Not needed, but could be considered a simplification by using an existing model?  I don't feel too strongly if this is what is boils down to, aside from API churn, but I guess better now that after release and more widespread use.
